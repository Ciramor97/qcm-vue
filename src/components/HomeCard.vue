<template>
  <div
    class="bg-pink px-[17px] py-[15px] rounded-lg shadow-custom flex flex-col items-center hover:bg-light-pink"
    @mouseover="isHovered = true"
    @mouseleave="isHovered = false"
  >
    <component :is="icon" :bgColor="isHovered ? hoverBgColor : bgColor" />
    <div
      class="mt-2 text-center p-1 font-bold text-sm"
      :class="
        isHovered ? 'text-light-pink bg-white' : 'text-white bg-light-pink'
      "
    >
      {{ title }}
    </div>
  </div>
</template>

<script setup lang="ts">
import { ref } from "vue";

defineProps<{
  icon: any;
  title: string;
  bgColor: string;
  hoverBgColor: string;
}>();
const isHovered = ref(false);
</script>
