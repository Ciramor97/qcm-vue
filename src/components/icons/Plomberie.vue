<template>
  <svg
    width="82"
    height="82"
    viewBox="0 0 82 82"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M58.0796 8.08557H50.1772C38.7888 8.08557 28.4341 12.5681 20.761 19.8517L22.8637 26.4396L30.1342 22.748L58.0796 8.08557ZM58.0796 8.08557V30.6676H52.1934C40.7526 30.6676 31.4453 39.9748 31.4453 51.4156V58.9991H7.40576V50.8566C7.40576 39.675 11.7268 29.4895 18.7779 21.8605L20.9288 28.5996L20.9289 28.5997C21.0405 28.9487 21.2992 29.2312 21.6373 29.3728L21.6382 29.3732C21.9756 29.513 22.3583 29.5008 22.6857 29.3337L22.6855 29.3339L22.5723 29.111L22.6859 29.3336L58.0796 8.08557ZM29.8377 25.7021L31.3183 32.1953L31.3183 32.1954C31.4507 32.7753 31.9659 33.1682 32.5364 33.1676M29.8377 25.7021L30.1344 22.7479C30.4781 22.5731 30.8832 22.5659 31.2317 22.7313C31.5807 22.8947 31.8332 23.2101 31.919 23.5844L31.919 23.5847L33.756 31.6399L33.756 31.6399C33.9092 32.3126 33.488 32.9827 32.815 33.1364M29.8377 25.7021L30.0676 25.5854L31.562 32.1398L29.8377 25.7021ZM29.8377 25.7021L29.7804 25.4508L29.8377 25.7021ZM32.5364 33.1676C32.5363 33.1676 32.5362 33.1676 32.5361 33.1676V32.9176L32.5364 33.1676ZM32.5364 33.1676C32.6278 33.1676 32.7211 33.1579 32.815 33.1364M32.815 33.1364C32.815 33.1364 32.815 33.1364 32.815 33.1364L32.7593 32.8927L32.815 33.1364ZM80 8.08557C80.6903 8.08557 81.25 7.52589 81.25 6.83557C81.25 6.14525 80.6903 5.58557 80 5.58557H71.708V5.16711C71.708 2.82348 69.8016 0.917114 67.458 0.917114H62.3296C59.986 0.917114 58.0796 2.82348 58.0796 5.16711V5.58557H50.1772C25.2145 5.58557 4.90576 25.8938 4.90576 50.8566V59.0002C2.60547 59.0505 0.75 60.937 0.75 63.2491V68.3776C0.75 70.6897 2.60547 72.5762 4.90576 72.6265V80.1671C4.90576 80.8574 5.46544 81.4171 6.15576 81.4171C6.84608 81.4171 7.40576 80.8574 7.40576 80.1671V72.6276H31.4453V80.1671C31.4453 80.8574 32.005 81.4171 32.6953 81.4171C33.3856 81.4171 33.9453 80.8574 33.9453 80.1671V72.6173C36.1639 72.4781 37.9268 70.631 37.9268 68.3776V63.2491C37.9268 60.9957 36.1639 59.1486 33.9453 59.0094V51.4156C33.9453 41.354 42.1317 33.1676 52.1934 33.1676H58.0796V33.8439C58.0796 36.1875 59.986 38.0939 62.3296 38.0939H67.458C69.8016 38.0939 71.708 36.1875 71.708 33.8439V33.1676H80C80.6903 33.1676 81.25 32.6079 81.25 31.9176C81.25 31.2273 80.6903 30.6676 80 30.6676H71.708V8.08557H80ZM35.4268 63.2491V68.3776C35.4268 69.3425 34.6417 70.1276 33.6768 70.1276H5C4.03504 70.1276 3.25 69.3425 3.25 68.3776V63.2491C3.25 62.2842 4.03504 61.4991 5 61.4991H33.6768C34.6417 61.4991 35.4268 62.2842 35.4268 63.2491ZM69.208 33.8439C69.208 34.8088 68.423 35.5939 67.458 35.5939H62.3296C61.3646 35.5939 60.5796 34.8088 60.5796 33.8439V5.16711C60.5796 4.20216 61.3646 3.41711 62.3296 3.41711H67.458C68.423 3.41711 69.208 4.20216 69.208 5.16711V33.8439Z"
      fill="#FF445F"
      stroke="#FF445F"
      stroke-width="0.5"
    />
  </svg>
</template>
