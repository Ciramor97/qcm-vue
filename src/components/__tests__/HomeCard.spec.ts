import { describe, expect, it } from "vitest";
import { mount } from "@vue/test-utils";
import HomeCard from "../HomeCard.vue";
import Plomberie from "../icons/Plomberie.vue";

describe("HomeCard Component", () => {
  const wrapper = mount(
    {
      template: "<HomeCard></HomeCard>",
    },
    {
      props: {
        icon: Plomberie,
        title: "Plomberie",
        bgColor: "FF445F",
        hoverBgColor: "#FFF",
      },

      global: {
        components: { HomeCard },
      },
    }
  );
  it("should renders", () => {
    expect(wrapper.exists()).toBe(true);
  });
  it("should match snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
