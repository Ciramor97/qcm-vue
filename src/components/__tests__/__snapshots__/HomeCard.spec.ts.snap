// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeCard Component > should match snapshot 1`] = `
"<div class="bg-pink px-[17px] py-[15px] rounded-lg shadow-custom flex flex-col items-center hover:bg-light-pink"><svg width="100" height="101" viewBox="0 0 100 101" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path d="M67.0796 17.0856H59.1772C47.7888 17.0856 37.4341 21.5681 29.761 28.8517L31.8637 35.4396L39.1342 31.748L67.0796 17.0856ZM67.0796 17.0856V39.6676H61.1934C49.7526 39.6676 40.4453 48.9748 40.4453 60.4156V67.9991H16.4058V59.8566C16.4058 48.675 20.7268 38.4895 27.7779 30.8605L29.9288 37.5996L29.9289 37.5997C30.0405 37.9487 30.2992 38.2312 30.6373 38.3728L30.6382 38.3732C30.9756 38.513 31.3583 38.5008 31.6857 38.3337L31.6855 38.3339L31.5723 38.111L31.6859 38.3336L67.0796 17.0856ZM38.8377 34.7021L40.3183 41.1953L40.3183 41.1954C40.4507 41.7753 40.9659 42.1682 41.5364 42.1676M38.8377 34.7021L39.1344 31.7479C39.4781 31.5731 39.8832 31.5659 40.2317 31.7313C40.5807 31.8947 40.8332 32.2101 40.919 32.5844L40.919 32.5847L42.756 40.6399L42.756 40.6399C42.9092 41.3126 42.488 41.9827 41.815 42.1364M38.8377 34.7021L39.0676 34.5854L40.562 41.1398L38.8377 34.7021ZM38.8377 34.7021L38.7804 34.4508L38.8377 34.7021ZM41.5364 42.1676C41.5363 42.1676 41.5362 42.1676 41.5361 42.1676V41.9176L41.5364 42.1676ZM41.5364 42.1676C41.6278 42.1676 41.7211 42.1579 41.815 42.1364M41.815 42.1364C41.815 42.1364 41.815 42.1364 41.815 42.1364L41.7593 41.8927L41.815 42.1364ZM89 17.0856C89.6903 17.0856 90.25 16.5259 90.25 15.8356C90.25 15.1453 89.6903 14.5856 89 14.5856H80.708V14.1671C80.708 11.8235 78.8016 9.91711 76.458 9.91711H71.3296C68.986 9.91711 67.0796 11.8235 67.0796 14.1671V14.5856H59.1772C34.2145 14.5856 13.9058 34.8938 13.9058 59.8566V68.0002C11.6055 68.0505 9.75 69.937 9.75 72.2491V77.3776C9.75 79.6897 11.6055 81.5762 13.9058 81.6265V89.1671C13.9058 89.8574 14.4654 90.4171 15.1558 90.4171C15.8461 90.4171 16.4058 89.8574 16.4058 89.1671V81.6276H40.4453V89.1671C40.4453 89.8574 41.005 90.4171 41.6953 90.4171C42.3856 90.4171 42.9453 89.8574 42.9453 89.1671V81.6173C45.1639 81.4781 46.9268 79.631 46.9268 77.3776V72.2491C46.9268 69.9957 45.1639 68.1486 42.9453 68.0094V60.4156C42.9453 50.354 51.1317 42.1676 61.1934 42.1676H67.0796V42.8439C67.0796 45.1875 68.986 47.0939 71.3296 47.0939H76.458C78.8016 47.0939 80.708 45.1875 80.708 42.8439V42.1676H89C89.6903 42.1676 90.25 41.6079 90.25 40.9176C90.25 40.2273 89.6903 39.6676 89 39.6676H80.708V17.0856H89ZM44.4268 72.2491V77.3776C44.4268 78.3425 43.6417 79.1276 42.6768 79.1276H14C13.035 79.1276 12.25 78.3425 12.25 77.3776V72.2491C12.25 71.2842 13.035 70.4991 14 70.4991H42.6768C43.6417 70.4991 44.4268 71.2842 44.4268 72.2491ZM78.208 42.8439C78.208 43.8088 77.423 44.5939 76.458 44.5939H71.3296C70.3646 44.5939 69.5796 43.8088 69.5796 42.8439V14.1671C69.5796 13.2022 70.3646 12.4171 71.3296 12.4171H76.458C77.423 12.4171 78.208 13.2022 78.208 14.1671V42.8439Z" fill="FF445F" stroke="FF445F" stroke-width="0.5"></path>
  </svg>
  <div class="mt-2 text-center p-1 font-bold text-sm text-white bg-light-pink">Plomberie</div>
</div>"
`;
